src/app/components/client/toolbar/UserMenu.tsx

```diff
@@ -4,24 +4,214 @@
 
 "use client";
 
+import { useRef } from "react";
+import Image from "next/image";
+import Link from "next/link";
 import { Session } from "next-auth";
-import { signIn, signOut } from "next-auth/react";
+import { signOut } from "next-auth/react";
+import { Item, useMenuTriggerState, useTreeState } from "react-stately";
+import { mergeProps, useMenuTrigger, useMenu, useMenuItem } from "react-aria";
 
-export type Props = {
+import type { AriaMenuOptions, AriaMenuTriggerProps } from "react-aria";
+import type { MenuTriggerProps, TreeState } from "react-stately";
+import type { Node } from "@react-types/shared";
+import type { ReactNode, Key } from "react";
+
+import { Popover } from "../Popover";
+import { SignInButton } from "../../../(nextjs_migration)/components/client/SignInButton";
+import { useL10n } from "../../../hooks/l10n";
+import styles from "./UserMenu.module.scss";
+import OpenInIcon from "./images/menu-icon-open-in.svg";
+import SettingsIcon from "./images/menu-icon-settings.svg";
+import HelpIcon from "./images/menu-icon-help.svg";
+import SignOutIcon from "./images/menu-icon-signout.svg";
+
+export type UserMenuProps = {
   user: Session["user"] | null;
 };
 
-export const UserMenu = (props: Props) => {
-  // Placeholder to enable login/logout until we create an actual menu
-  // Ignored for test coverage; to be implemented:
-  /* c8 ignore next 3 */
+export const UserMenu = (props: UserMenuProps) => {
+  const l10n = useL10n();
+
+  const fxaItemRef = useRef<HTMLAnchorElement>(null);
+  const settingsItemRef = useRef<HTMLAnchorElement>(null);
+  const helpItemRef = useRef<HTMLAnchorElement>(null);
+  const signOutItemRef = useRef<HTMLButtonElement>(null);
+
   if (!props.user) {
-    return <button onClick={() => void signIn("fxa")}>Sign in</button>;
+    return <SignInButton />;
   }
 
+  const itemKeys = {
+    fxa: "fxa",
+    settings: "settings",
+    help: "help",
+    signout: "signout",
+  };
+
+  const handleOnAction = (menuItemKey: Key) => {
+    switch (menuItemKey) {
+      case itemKeys.fxa:
+        fxaItemRef.current?.click();
+        break;
+      case itemKeys.settings:
+        settingsItemRef.current?.click();
+        break;
+      case itemKeys.help:
+        helpItemRef.current?.click();
+        break;
+      case itemKeys.signout:
+        signOutItemRef.current?.click();
+        break;
+      default:
+      // do nothing
+    }
+  };
+
   return (
-    // Ignored for test coverage; to be implemented:
-    /* c8 ignore next */
-    <button onClick={() => void signOut({ callbackUrl: "/" })}>Sign out</button>
+    <MenuTrigger user={props.user} onAction={handleOnAction}>
+      <Item
+        key={itemKeys.fxa}
+        textValue={l10n.getString("user-menu-manage-fxa-label")}
+      >
+        <b>{props.user.email}</b>
+        <a
+          className={styles.menuItemCta}
+          href={process.env.NEXT_PUBLIC_FXA_SETTINGS_URL}
+          ref={fxaItemRef}
+          target="_blank"
```

Comment:
Does this need `rel="noopener noreferrer"` like the link below (L110)?

Reply:
Missed it and added in [892432c](https://github.com/mozilla/blurts-server/pull/3326/commits/892432c3c0fa0502ae056a625c8dd387c498e629).